src/olympia/reviewers/models.py

```diff
@@ -565,29 +565,28 @@ def check_has_auto_approval_disabled(cls, version):
         """Check whether the add-on has auto approval disabled or delayed.
 
         It could be:
-        - Disabled by a reviewer (different flag for listed or unlisted)
-        - Disabled until next manual approval (only applies to listed, typically
-          set when a previous version is on a delayed rejection)
-        - Delayed until a future date by scanners (applies to both listed and
-          unlisted)
+        - Disabled
+        - Disabled until next manual approval
+        - Delayed until a future date
+
+        Those flags are set by scanners or reviewers for a specific channel.
         """
         addon = version.addon
-        is_listed = version.channel == amo.CHANNEL_LISTED
-        if is_listed:
-            auto_approval_disabled = bool(
-                addon.auto_approval_disabled
-                or addon.auto_approval_disabled_until_next_approval
-            )
-        else:
-            auto_approval_disabled = bool(
-                addon.auto_approval_disabled_unlisted
-                or addon.auto_approval_disabled_until_next_approval_unlisted
-            )
-        auto_approval_delayed = bool(
-            addon.auto_approval_delayed_until
-            and datetime.now() < addon.auto_approval_delayed_until
+        flag_suffix = '_unlisted' if version.channel == amo.CHANNEL_UNLISTED else ''
+        auto_approval_disabled = bool(
+            getattr(addon, f'auto_approval_disabled{flag_suffix}')
+        )
+        auto_approval_disabled_until_next_approval = getattr(
+            addon, f'auto_approval_disabled_until_next_approval{flag_suffix}'
+        )
+        auto_approval_delayed = getattr(
```

Comment:
Nit: This sounds like a boolean to me, but I think it's actually a date, so maybe change the name to make that more obvious?

Reply:
https://github.com/mozilla/addons-server/pull/20267/commits/4060db1d8815831942342dc3bef32bfe7522c835 and https://github.com/mozilla/addons-server/pull/20267/commits/21b46d320e16984622d581258a4331410c2bf96d should help a bit